test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/DependencyGraphSpecGeneratorTests.cs

```diff
@@ -0,0 +1,705 @@
+        public void DependencyGraphSpecGenerator_GetOriginalTargetFrameworks_WhenTargetFramworksNotSpecified(string targetFrameworks, string[] expected)
+        {
+            var project = new MockMSBuildProject(new Dictionary<string, string>
+            {
+                ["TargetFrameworks"] = null
+            });
+
+            var actual = MSBuildStaticGraphRestore.GetOriginalTargetFrameworks(project, targetFrameworks.Split(';').Select(i => NuGetFramework.Parse(i)).ToList());
+
+            actual.ShouldBeEquivalentTo(expected);
+        }
+
+        [Theory]
+        [InlineData("net45", new[] { "net45" })]
+        [InlineData("net40;net45", new[] { "net40", "net45" })]
+        [InlineData("net40;net45 ; netstandard2.0 ", new[] { "net40", "net45", "netstandard2.0" })]
+        public void DependencyGraphSpecGenerator_GetOriginalTargetFrameworks_WhenTargetFramworksSpecified(string targetFrameworks, string[] expected)
+        {
+            var project = new MockMSBuildProject(new Dictionary<string, string>
+            {
+                ["TargetFrameworks"] = targetFrameworks
+            });
+
+            var actual = MSBuildStaticGraphRestore.GetOriginalTargetFrameworks(project, new List<NuGetFramework>());
+
+            actual.ShouldBeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void DependencyGraphSpecGenerator_GetPackageDownloads()
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var project = new MockMSBuildProject(testDirectory)
+                {
+                    Items = new Dictionary<string, IList<IMSBuildItem>>
+                    {
+                        ["PackageDownload"] = new List<IMSBuildItem>
+                        {
+                            new MSBuildItem("PackageA", new Dictionary<string, string> { ["Version"] = "[1.1.1]" }),
+                            new MSBuildItem("PackageA", new Dictionary<string, string> { ["Version"] = "[2.0.0]" }),
+                            new MSBuildItem("PackageB", new Dictionary<string, string> { ["Version"] = "[1.2.3];[4.5.6]" }),
+                        }
+                    }
+                };
+
+                var actual = MSBuildStaticGraphRestore.GetPackageDownloads(project);
+
+                actual.ShouldBeEquivalentTo(new List<DownloadDependency>
+                {
+                    new DownloadDependency("PackageA", VersionRange.Parse("[1.1.1]")),
+                    new DownloadDependency("PackageB", VersionRange.Parse("[1.2.3]")),
+                    new DownloadDependency("PackageB", VersionRange.Parse("[4.5.6]")),
+                });
+            }
+        }
+
+        [Theory]
+        [InlineData("1.2.3")]
+        [InlineData("(1.2.3,]")]
+        [InlineData("1.*")]
+        [InlineData("[1.2.3];4.5.6", "4.5.6")]
+        public void DependencyGraphSpecGenerator_GetPackageDownloads_ThrowsWhenNotExactVersion(string version, string expected = null)
```

Comment:
Rename tests:

    <MemberName>_<Condition>__<ExpectedResult>

This would be:

    GetPackageDownloads_WhenNotExactVersion_Throws

The `DependencyGraphSpecGenerator` name is obsolete.